### ISSUE TITLE
`sphinx-build -n -W` doesn't treat nitpicky warnings as errors 

### ISSUE THREAD

**Issue by goosemania** [sphinx-doc/sphinx#3919]
### Problem
In sphinx 1.5 when [`-n` (nitpicky mode)](http://www.sphinx-doc.org/en/stable/invocation.html#cmdoption-sphinx-build-n) and [`-W` (treat warnings as errors)](http://www.sphinx-doc.org/en/stable/invocation.html#id6) are used together, all the warnings including the ones which come from nitpicky mode will throw an error and fail a build. That's the expected behavior.

It's no longer a case in 1.6, all the warnings for missing references are left as just warnings and they don't fail a build, even with `-W` flag.

### Environment info
- OS: Fedora 25
- Python version: 3.5.3
- Sphinx version: 1.6.3

**Comment by TimKam** [comment]
I tried to reproduce this, but couldn't. Can you point to/provide an example project?

**Comment by goosemania** [comment]
@TimKam , thanks for looking into it!
Right now I can provide a link to the project where the issue can be seen, I hope it's not too complicated for investigation. I can try to create a smaller reproducer but later (in several weeks). But with the following project it is easily reproducible, sphinx 1.5 works as expected, 1.6 - does not convert warnings to errors. 

The affected project:
`git clone -b 3.0-dev https://github.com/pulp/pulp.git`

- [Makefile](https://github.com/pulp/pulp/blob/3.0-dev/docs/Makefile#L5) where `-n` should be added to observe this issue.
- [Sphinx config](https://github.com/pulp/pulp/blob/3.0-dev/docs/conf.py), just in case there is something obvious missing.
- [Some dependencies and config file](https://github.com/pulp/pulp_packaging/blob/7adce8de7735e0c778eaf697cd29f7a72080c2db/ci/jjb/jobs/pr-docs.yaml#L73-L82) which are needed to build docs

**Comment by TimKam** [comment]
The setup is fairly complex and ad-hoc I can't find any obvious issue.
If you provide a smaller docs example that doesn't require a full Django setup, I'll take another look asap.

**Comment by jschueller** [comment]
@goosemania maybe you could try to bisect the problem to find the commit that broke this.

**Comment by jschueller** [comment]
hello @goosemania,
I just tested your example, nitpick works well here in both 1.5.5 and 1.6.3, so I cannot reproduce the issue.

Here's the script to build it (ubuntu 16.04): 
```
#!/bin/sh

cd /tmp
if ! test -d sphinx
then
  git clone https://github.com/sphinx-doc/sphinx.git
fi
cd sphinx
git checkout 1.5.6
# git checkout 1.6.3
rm -rf ~/.local/lib/python3.5/sphinx* ~/.local/lib/python3.5/Sphinx*
python3 setup.py install --user
cd ..

if ! test -d pulp
then
  git clone https://github.com/pulp/pulp.git
fi
cd pulp
cd docs
make html SPHINXBUILD=/usr/bin/sphinx-build3 SPHINXOPTS="-W -n"
```

**Comment by tsouvarev** [comment]
Seems like Django somehow intercepting logging, thus creating this issue.

Project to reproduce: https://github.com/tsouvarev/sphinx-broken-nitpicky-example

If run `make html` standing on the latest commit, you won't get any errors, thou `source/index.rst` contains broken reference. If you comment [lines 138-142 in nitpicky/settings.py](https://github.com/tsouvarev/sphinx-broken-nitpicky-example/blob/master/nitpicky/settings.py#L138-L142), you will finally get compilation error.

**Comment by tk0miya** [comment]
@tsouvarev Thank you for your help. I understand the reason of this problem.

Since Sphinx-1.6, Sphinx uses standard logging module internally to show all messages including errors and warnings. But the settings of these apps retets its settings. So all messages are not shown and ignored.

I'll continue to investigate this.

**Comment by tk0miya** [comment]
Fixed by #4180 
Thank you for reporting
